jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/ProcessHandler.java

```diff
@@ -461,7 +461,8 @@ private static void linkBoundaryTimerEvent(NodeContainer nodeContainer, Node nod
         String timeCycle = (String) node.getMetaData().get("TimeCycle");
         String timeDate = (String) node.getMetaData().get("TimeDate");
         Timer timer = new Timer();
-        timer.setName(node.getName());
+        String boundaryEventName = node.getName();
+        timer.setName(compositeNode.getName() +(boundaryEventName != null ? "-"+boundaryEventName : ""));
```

Comment:
if human task has no name (`compositeNode.getName()` will return "null"): 
Perhaps, in this scenario, it's better to assign "" (as well as logging a WARN trace)

Reply:
This scenario has been taken into account